### PR TITLE
Init API for search history

### PR DESCRIPTION
**Pull request by thaohq-0814**
Init API for search history
Link: https://edu-redmine.framgia.vn/issues/6937

### CODE REVIEW

**Comment by thaohq-0814** [comment]
ready

File: app/src/main/java/com/framgia/my_editor_02/screen/collections/CollectionsFragment.java [hunk 1]
```diff
@@ -33,7 +35,8 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                 DataBindingUtil.inflate(inflater, R.layout.fragment_collections, container, false);
         ImageRepository repository =
                 ImageRepository.getInstance(ImageRemoteDataSource.getInstance(),
-                        ImageLocalDataSource.getInstance());
+                        ImageLocalDataSource.getInstance(
+                                new SharedPrefsApi(Objects.requireNonNull(getContext()))));
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/my_editor_02/screen/collections/CollectionsFragment.java]: using applicationContext

File: app/src/main/java/com/framgia/my_editor_02/screen/photos/PhotosFragment.java [hunk 2]
```diff
@@ -63,7 +64,8 @@ private void setUp() {
         mNavigator = new Navigator(this);
         ImageRepository repository =
                 ImageRepository.getInstance(ImageRemoteDataSource.getInstance(),
-                        ImageLocalDataSource.getInstance());
+                        ImageLocalDataSource.getInstance(
+                                new SharedPrefsApi(Objects.requireNonNull(getContext()))));
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/my_editor_02/screen/photos/PhotosFragment.java]: as above

File: app/src/main/java/com/framgia/my_editor_02/screen/search/SearchFragment.java [hunk 3]
```diff
@@ -52,7 +54,8 @@ public void onStop() {
     private void setUp() {
         ImageRepository repository =
                 ImageRepository.getInstance(ImageRemoteDataSource.getInstance(),
-                        ImageLocalDataSource.getInstance());
+                        ImageLocalDataSource.getInstance(
+                                new SharedPrefsApi(Objects.requireNonNull(getContext()))));
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/my_editor_02/screen/search/SearchFragment.java]: as above

**Comment by thaohq-0814** [comment]
ready

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**